send-transaction-service/src/send_transaction_service.rs

```diff
@@ -360,34 +171,41 @@ impl SendTransactionService {
         bank_forks: &Arc<RwLock<BankForks>>,
         leader_info: Option<T>,
         receiver: Receiver<TransactionInfo>,
-        connection_cache: &Arc<ConnectionCache>,
+        connection_cache: Arc<ConnectionCache>,
         config: Config,
         exit: Arc<AtomicBool>,
     ) -> Self {
         let stats_report = Arc::new(SendTransactionServiceStatsReport::default());
 
         let retry_transactions = Arc::new(Mutex::new(HashMap::new()));
 
-        let leader_info_provider = Arc::new(Mutex::new(CurrentLeaderInfo::new(leader_info)));
+        let client = ConnectionCacheClient::new(
+            connection_cache,
+            tpu_address,
+            config.tpu_peers,
+            leader_info,
+            config.leader_forward_count,
+        );
 
         let receive_txn_thread = Self::receive_txn_thread(
-            tpu_address,
             receiver,
-            leader_info_provider.clone(),
-            connection_cache.clone(),
-            config.clone(),
+            client.clone(),
```

Comment:
we clone the client, which implies cloning of the vector of sockets. The same happen before but inside config.clone(), so no new memory allocations have been introduced.